src/tiled/resizehelper.cpp

```diff
@@ -118,14 +130,17 @@ void ResizeHelper::paintEvent(QPaintEvent *)
     pen.setCosmetic(true);
 
     painter.setPen(pen);
-    painter.drawRect(QRect(QPoint(0, 0), mNewSize));
+    painter.drawRect(QRect(newRectOffset, mNewSize));
 
     pen.setColor(Qt::white);
 
-    painter.setPen(pen);
-    painter.setBrush(Qt::white);
     painter.setOpacity(0.5);
-    painter.drawRect(oldRect);
+    QImage minimap = QImage(oldRect.width() * mScale, oldRect.height() * mScale, QImage::Format_ARGB32_Premultiplied);
+    mMiniMapRenderer->renderMinimapToImage(minimap, MiniMapRenderer::DrawObjects
+                                           | MiniMapRenderer::DrawImages
+                                           | MiniMapRenderer::DrawTiles
+                                           | MiniMapRenderer::IgnoreInvisibleLayer);
```

Comment:
Rendering the mini-map on each paint event is too slow for large maps. Instead, please keep the image around and only repaint it when the scale changes.

Ideally, we'd do the rendering in a thread. I think we can easily do that here, because the resize helper dialog is modal so there can't be any changes to the underlying data in the meantime. Up to you whether you want to look into that or leave it for later.